src/components/charts/sizeDistancePlotter/LegendMultiple.jsx

```diff
@@ -7,21 +7,22 @@ import styles from './size-distance-plotter.module.scss';
 
 class LegendMultiple extends React.PureComponent {
   render() {
-    const { yValueAccessor, numOfSets } = this.props;
+    const { yValueAccessor, numOfSets, data } = this.props;
     const useFluxColor = yValueAccessor === 'color';
 
     return (
       <Card className={styles.legend}>
-        {numOfSets &&
-          Array.from(Array(numOfSets)).map((set, i) => {
+        {data &&
+          numOfSets &&
+          data.map((datum, i) => {
             const isDefaultSet = i === 0;
             const setId = `set-${i}`;
             const pointClasses = classnames(
               styles.legendPoint,
               setId,
               styles.groupPoint,
+              styles[`colorSet${i}`],
```

Comment:
I think we probably only want the colorful colors used so let's do `i + 1` to skip over the grey